src/Public/js/nipsweb.player.js

```diff
@@ -1,685 +1,1105 @@
+            type: 'POST',
+            success: function(data) {
+                clientid = parseInt(data.token);
+            },
+            async: false
+        });
+    }
+
+    /**
+    * Returns number of minutes (zero padded) from a time in seconds
+    * @param time in seconds
+    */
+    var timeMins = function(time) {
+        var mins = Math.floor(time / 60) + "";
+        if (mins.length < 2) {
+            mins = '0' + mins;
+        }
+        return mins;
+    };
+
+    // Returns number of seconds (zero padded) less than mins from a time in seconds
+    var timeSecs = function(time) {
+        var secs = Math.floor(time % 60) + "";
+        if (secs.length < 2) {
+            secs = '0' + secs;
+        }
+        return secs;
+    };
+
+    // Gets the time of the current track in channel
+    var getTime = function(channel) {
+        var audio = getPlayer(channel);
+
+        var elapsed = audio.currentTime; //Get the current playing position of the track
+        //currentTime returns a value in seconds. Convert to minutes+seconds, pad zeros where appropriate.
+        var minelap = timeMins(elapsed);
+        var secelap = timeSecs(elapsed);
+        // Sets the current time label
+        $('#ch' + channel + '-elapsed').html(minelap + ':' + secelap);
+    }
+
+    // Gets the duration of the current track in channel
+    var getDuration = function(channel) {
+        var audio = getPlayer(channel);
+
+        var duration = audio.duration; //Get the duration of the track
+        //duration returns a value in seconds. Convert to minutes+seconds, pad zeros where appropriate.
+        var mindur = timeMins(duration);
+        var secdur = timeSecs(duration);
+        // Sets the duration label
+        $('#ch' + channel + '-duration').html(mindur + ':' + secdur);
+    };
+
+    var showAlert = function(text, type) {
+        if (!type) {
+            type = 'success';
+        }
+
+        var close = '<button type="button" class="close" data-dismiss="alert"><span aria-hidden="true">×</span><span class="sr-only">Close</span></button>';
+
+        var alert = $('<div></div>').addClass('footer-alert').addClass('alert').addClass('alert-'+type).html(text + close);
+        alert.alert();
+
+        setTimeout(function() {alert.alert('close')}, 15000);
+
+        $(document.body).append(alert);
+    }
+
+    /**
+    * Change shipping operates in a queue - this ensures that changes are sent atomically and sequentially.
+    * ops: JSON changeset to send
+    * addOp: If true, there has been an add operation. We currently make these syncronous.
+    * pNext: Optional. Parent queue to process on completion.
+    */
+    var shipChanges = function(ops, addOp, pNext) {
+        if (typeof addOp === 'undefined') {
+            addOp = false;
+        }
+
+        ajaxQueue.queue(function(next) {
+            $('#notice').html('Saving changes...').show();
+            $.ajax({
+                async: !addOp,
```

Comment:
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check http://xhr.spec.whatwg.org/.